driver/driver.go

```diff
@@ -0,0 +1,169 @@
+// Implemented according to
+// https://github.com/docker/libnetwork/blob/master/docs/remote.md
+
+package driver
+
+import (
+	"fmt"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/go-plugins-helpers/network"
+)
+
+const (
+	// DriverName is name of the driver that is to be specified during docker network creation
+	DriverName = "Contrail"
+
+	// HNSNetworkName is a constant name for HNS network
+	NetworkHNSname = "ContrailHNSNet"
+)
+
+type ContrailDriver struct {
+	HnsID string
+}
+
+func NewDriver() (*ContrailDriver, error) {
+
+	subnets := []hcsshim.Subnet{
+		{
+			AddressPrefix:  "172.117.0.0/16",
+			GatewayAddress: "172.117.0.1",
+		},
+	}
+
+	configuration := &hcsshim.HNSNetwork{
+		Name:               NetworkHNSname,
+		Type:               "transparent",
+		Subnets:            subnets,
+		NetworkAdapterName: "Ethernet0",
+	}
+
+	hnsID, err := CreateHNSNetwork(configuration)
+	if err != nil {
+		return nil, err
+	}
+
+	d := &ContrailDriver{
+		HnsID: hnsID,
+	}
+	return d, nil
+}
+
+func (d *ContrailDriver) Teardown() error {
+	err := DeleteHNSNetwork(d.HnsID)
```

Comment:
HNS log requests as using debugln in hns.go, errors are caught and logged in func main. You mean something different?